back/engines/commercial/verification/app/services/verification/patches/permissions/user_requirements_service.rb

```diff
@@ -9,22 +9,26 @@ module UserRequirementsService
         def base_requirements(permission)
           requirements = super
 
-          if VerificationService.new.find_verification_group(permission.groups)
+          if verification_service.find_verification_group(permission.groups)
             requirements[:special][:verification] = 'require'
           end
           requirements
         end
 
         def mark_satisfied_requirements!(requirements, permission, user)
           super
-          return unless permission.permitted_by == 'groups' && VerificationService.new.find_verification_group(permission.groups)
+          return unless permission.permitted_by == 'groups' && verification_service.find_verification_group(permission.groups)
 
           if user.verified?
             requirements[:special][:verification] = 'satisfied'
           elsif user.in_any_groups? permission.groups
             requirements[:special][:verification] = 'dont_ask'
           end
         end
+
+        def verification_service
+          @verification_service ||= VerificationService.new
+        end
```

Comment:
Not sure this makes any difference in what I've been testing but it might in future PRs